File: app/src/main/java/com/mypackage/it314_health_center/startups/pres_adapter.java
package com.mypackage.it314_health_center.startups;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.mypackage.it314_health_center.R;

import java.util.List;

public class pres_adapter extends RecyclerView.Adapter<pres_adapter.ViewHolder> {

    private List<prescription_class> pres_list;

    public pres_adapter (List<prescription_class>pres_list){this.pres_list=pres_list;}

    @NonNull
    @Override
    public pres_adapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_design,parent,false);
        return new pres_adapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull pres_adapter.ViewHolder holder, int position) {
        int resource = pres_list.get(position).getPres_id1();
        String date = pres_list.get(position).getPres_text_id1();
        String name = pres_list.get(position).getPres_text_id2();
        String line = pres_list.get(position).getDivider();

        holder.setData(resource, date, name, line);
    }

    @Override
    public int getItemCount() {
        return pres_list.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder{

        private ImageView imageView;
        private TextView textView;
        private TextView textView2;
        private TextView divider;


        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            imageView = itemView.findViewById(R.id.pres_id1);
            textView = itemView.findViewById(R.id.pres_text_id1);
            textView = itemView.findViewById(R.id.pres_text_id2);
            textView = itemView.findViewById(R.id.divider);
        }

        public void setData(int resource, String date, String name, String line) {
            imageView.setImageResource(resource);
            textView.setText(date);
            textView2.setText(name);
            divider.setText(line);
        }
    }
}

File: app/src/main/java/com/mypackage/it314_health_center/patient_side/Pres_adapter.kt
package com.mypackage.it314_health_center.patient_side

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.mypackage.it314_health_center.R
import com.mypackage.it314_health_center.models.prescription_class
import java.util.*

class pres_adapter(private val pres_list: List<prescription_class>) :
    RecyclerView.Adapter<pres_adapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.prescriptions_content, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val pos=pres_list[position]
        val pres_img = pres_list[position].image_url
        val issued_by = pres_list[position].issued_by
        val date = pres_list[position].date
        holder.setData(date, issued_by)
    }

    override fun getItemCount(): Int {
        return pres_list.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private var pres_image_id: ImageView? = null
        private var date_id: TextView? = null
        private var time_id: TextView? = null
        private var issue_id: TextView? = null

        init {
            pres_image_id = itemView.findViewById(R.id.pres_image_id)
            date_id = itemView.findViewById(R.id.date_id)
            time_id = itemView.findViewById(R.id.time_id)
            issue_id = itemView.findViewById(R.id.issue_id)
        }

        fun setData(date: Date, name: String) {

        }
    }
}